src/Platforms/OMPTarget/OMPallocator.hpp

```diff
@@ -34,6 +35,22 @@ T* getOffloadDevicePtr(T* host_ptr)
   return device_ptr;
 }
 
+/** Return the device ptr for memory inside of the dual memory allocation.
+ *  Since we have containers of containers memory and dual memory spaces we need this.
+ */
+template<typename T>
+T* getOffloadDevicePtr(T* allocator_ptr, T* host_ptr)
```

Comment:
https://github.com/QMCPACK/miniqmc/blob/OMP_offload/src/Platforms/tests/OMPTarget/test_omp_allocator.cpp
I don't think we need function

Reply:
We defintely need this because I want the API to include that actual data dependence of the computation.  I don't want to use VectorSoAContainer object as an argument but a Matrix but a matrix made from a part of a psiM_vgl as psiV_temp is has a broken allocator that can not provide a device pointer. I should not need to know about the allocator returned pointer that the owned object has only the pointer it was constructed or attached with.

Reply:
I probably don't understand what your real need is.

Reply:
I have a integration test between OhmmsMatrix, VectorSoAContainer, OMPAllocator just about done that will hopefully clarify this.

Reply:
Correct you are @ye-luo.
nonsense function removed.